src/services/event/event-router.ts

```diff
@@ -320,17 +320,25 @@ eventsRouter.get("/:EVENTID/", weakJwtVerification, async (req: Request, res: Re
 eventsRouter.get("/", weakJwtVerification, async (_: Request, res: Response) => {
     const payload: JwtPayload = res.locals.payload as JwtPayload;
 
-    // Get collection from the database, and return it as an array
-    const publicEvents: Event[] = await Models.Event.find({ isStaff: false });
-
-    if (hasStaffPerms(payload)) {
-        return res.status(StatusCode.SuccessOK).send({ events: publicEvents });
+    let publicEvents: Event[];
+    if (payload === undefined) {
+        console.log("no jwt token");
+        publicEvents = await Models.Event.find({ isStaff: false, isPrivate: false });
```

Comment:
you can get rid of this and include this in the else condition

Reply:
actually, i'd add a check for isAttendee, that checks if the role is attendee and then use that